### PR TITLE
SPCTR-2347 - Global: Font not loading in the editor when using FSE theme

### PR DESCRIPTION
**Pull request by supriyajjadhav**
### Description
 Test Spectra blocks with Site Editor theme and report bugs.

### Types of changes
Bug fix (non-breaking change which fixes an issue) 

### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: src/components/typography/fontloader.js [hunk 1]
```diff
@@ -59,6 +59,17 @@ const WebfontLoader = ( props ) => {
 			} );
 			addFont( props.config.google.families[ 0 ] );
 		}
+		const siteEditTheme = document.getElementsByClassName( 'edit-site' );
+		
+		if( siteEditTheme.length !== 0 ) {
```
Comment (RajkiranBagal) [inline src/components/typography/fontloader.js:64]: Check condition for iframe present too, like below

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

File: src/components/typography/fontloader.js (hunk 1 above)
Comment (supriyajjadhav) [reply src/components/typography/fontloader.js]: Resolved.

File: src/components/typography/fontloader.js [hunk 2]
```diff
@@ -67,16 +78,16 @@ const WebfontLoader = ( props ) => {
...
-			if ( iframe ) {
+			if ( responsiveIframe ) {
 
 				WebFont.load( {
 					...props.config,
 					loading: handleLoading,
 					active: handleActive,
 					inactive: handleInactive,
-					context: iframe?.contentWindow
+					context: responsiveIframe?.frameElement?.contentWindow
```
Comment (rahulvarma722) [inline src/components/typography/fontloader.js:90]: If responsiveIframe?.frameElement?.contentWindow = undefined then how can we manage please add proper condition here

File: src/components/typography/fontloader.js [hunk 3]
```diff
@@ -59,6 +59,17 @@ const WebfontLoader = ( props ) => {
...
 		}
+		const siteEditTheme = document.getElementsByClassName( 'edit-site' );
+		
+		if( siteEditTheme && siteEditTheme.length !== 0 ) {
+			WebFont.load( {
+				...props.config,
+				loading: handleLoading,
+				active: handleActive,
+				inactive: handleInactive,
+				context: window.frames[0]?.frameElement?.contentWindow,
```
Comment (rahulvarma722) [inline src/components/typography/fontloader.js:70]: If window.frames[0]?.frameElement?.contentWindow = undefined then how can we manage please add proper condition here

File: src/components/typography/fontloader.js (hunk 2 above)
Comment (supriyajjadhav) [reply src/components/typography/fontloader.js]: @rahulvarma722 I have updated the comment please check

File: src/components/typography/fontloader.js (hunk 3 above)
Comment (supriyajjadhav) [reply src/components/typography/fontloader.js]: @rahulvarma722 I have updated the comment please check

**Review by rahulvarma722 — APPROVED** [review]

**Review by rahulvarma722 — APPROVED** [review]

**Review by AbhijitNage123 — APPROVED** [review]

**Review by RajkiranBagal — APPROVED** [review]

**Pull request merged**